### PR TITLE
Validate response examples to contain all example props of the schema…

### PR DESCRIPTION
**Pull request by Aboruhen**
… (#776)

- B014 - additional rule for example validation

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -15,6 +15,10 @@ It currently consists of
 
 # Release Notes
 BOAT is still under development and subject to change.
+## 0.17.32
+* add new link rule `B014` - to validate if examples contain all defined properties in schema
```
Comment (jjjasper) [inline README.md:19]: Type 'lint' ?

Comment (Aboruhen) [reply README.md]: fixed

File: boat-quay/boat-quay-rules/src/main/kotlin/com/backbase/oss/boat/quay/ruleset/RequestResponseExampleRule.kt [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    @Check(severity = Severity.SHOULD)
+    fun checkResponseExampleFulfill(context: Context): List<Violation> =
+            context.validateOperations { (method, operation) ->
+                operation?.responses.orEmpty()
+                        .map { (status, response) ->
+                            response.content.orEmpty()
+                                    .filterNot { (_, content) ->
+                                        hasDefinedAllParams(content)
+                                    }.map { (type, content) ->
+                                        context.violation("Fulfill the response example for ${method.name}:$status for $type to include all properties ", content)
```
[the remark this answers is not in the document]
Comment (Aboruhen) [reply boat-quay/boat-quay-rules/src/main/kotlin/com/backbase/oss/boat/quay/ruleset/RequestResponseExampleRule.kt]: updated

File: boat-quay/boat-quay-rules/src/main/kotlin/com/backbase/oss/boat/quay/ruleset/PluralizeResourceNamesRule.kt [hunk 3]
```diff
@@ -12,7 +12,7 @@ import org.zalando.zally.rule.api.*
 @Rule(
         ruleSet = BoatRuleSet::class,
         id = "B010",
-        severity = Severity.HINT,
+        severity = Severity.MUST,
```
Comment (jjjasper) [inline boat-quay/boat-quay-rules/src/main/kotlin/com/backbase/oss/boat/quay/ruleset/PluralizeResourceNamesRule.kt:15]: This change seems to be outside of the scope of this PR and together with the change in https://github.com/Backbase/backbase-openapi-tools/pull/779/files#diff-0d37ac6bc75b575f896b44ed22963ae103450faad888a8dd3ce346bcafd8af16R59 forces teams to fix *now* which not possible because changing the resource name is a breaking change.

Comment (Aboruhen) [reply boat-quay/boat-quay-rules/src/main/kotlin/com/backbase/oss/boat/quay/ruleset/PluralizeResourceNamesRule.kt:15]: In the scope of this change, I made the same behavior. According to old SEVERITY, this rule was a HINT, but our LintMojo didn't check for severity levels, so any rule mismatch requires a fix. And it's not strict for a new Rule B014(Examples check).

Comment (jjjasper) [reply boat-quay/boat-quay-rules/src/main/kotlin/com/backbase/oss/boat/quay/ruleset/PluralizeResourceNamesRule.kt]: Sorry, I think the github UI was playing tricks on me... it looked to me like this change was on another rule.

Comment (Aboruhen) [reply boat-quay/boat-quay-rules/src/main/kotlin/com/backbase/oss/boat/quay/ruleset/PluralizeResourceNamesRule.kt]: agree, about reading logs until fail.
I will improve the error by adding a missed property.
Additionally, should we check only 2xx or others as well?

Comment (jjjasper) [reply boat-quay/boat-quay-rules/src/main/kotlin/com/backbase/oss/boat/quay/ruleset/PluralizeResourceNamesRule.kt]: Most other response codes will be 4xx and 5xx (by the way: our guideline says to not define the 5xx because infra may throw these at the client anyway. Without the expected Error object...) but if you add this rule, then it should be for all examples. Can you also cover the POST/PUT request payloads?

**Comment by santam85** [comment]
@jjjasper @Aboruhen what's the status of this? Ready to merge?

File: boat-quay/boat-quay-rules/src/main/kotlin/com/backbase/oss/boat/quay/ruleset/PluralizeResourceNamesRule.kt (hunk 3 above)
Comment (Aboruhen) [reply boat-quay/boat-quay-rules/src/main/kotlin/com/backbase/oss/boat/quay/ruleset/PluralizeResourceNamesRule.kt]: done

File: README.md [hunk 4]
```diff
@@ -16,6 +16,8 @@ It currently consists of
 # Release Notes
 BOAT is still under development and subject to change.
 ## 0.17.34
+* add new lint rule `B014` - to validate if examples contain all defined properties in schema
```
Comment (santam85) [inline README.md:19]: New dev version is 0.17.35 please update

Comment (Aboruhen) [reply README.md:19]: The changes were on version 34. But none updated the readme so there was no conflict.
